OpenRA.Mods.Common/Traits/World/EditorHighlightLayer.cs

```diff
@@ -0,0 +1,81 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Required for the map editor to work. Attach this to the world actor.")]
+	public class EditorHighlightLayerInfo : ITraitInfo
+	{
+		[PaletteReference]
+		[Desc("Palette to use for rendering the placement sprite.")]
+		public readonly string Palette = TileSet.TerrainPaletteInternalName;
+
+		[Desc("Sequence image where the selection overlay types are defined.")]
+		public readonly string Image = "editor-overlay";
+
+		[SequenceReference("Image")]
+		[Desc("Sequence to use for the highlight overlay.")]
+		public readonly string HighlightSequence = "highlight";
+
+		public virtual object Create(ActorInitializer init) { return new EditorHighlightLayer(init.Self, this); }
+	}
+
+	public class EditorHighlightLayer : IWorldLoaded, IPostRender
```

Comment:
I couldn't find a good way to merge them together, so I extracted the shared logic to a parent class, is this ok?